azurelinuxagent/common/AgentGlobals.py

```diff
@@ -28,10 +82,39 @@ class AgentGlobals(object):
     #
     _container_id = "00000000-0000-0000-0000-000000000000"
 
+    # Features that we need to report to CRP in the status blob if we support them (Eg: MultiConfig)
+    __crp_supported_features = {
+        FeatureNames.MultiConfig: _MultiConfigFeature()
+    }
+
+    # Features that we use internally in the Guest Agent (Eg: ExtensionTelemetryPipeline)
+    __internal_features = {
+        FeatureNames.ExtensionTelemetryPipeline: _ExtensionTelemetryPipelineFeature()
+    }
+
+    @staticmethod
+    def get_feature_by_name(feature_name):
+        if feature_name in AgentGlobals.__internal_features:
+            return AgentGlobals.__internal_features[feature_name]
+
+        if feature_name in AgentGlobals.__crp_supported_features:
+            return AgentGlobals.__crp_supported_features[feature_name]
+
+        raise NotImplementedError("Feature with Name: {0} not found".format(feature_name))
+
+    @staticmethod
+    def get_crp_supported_features():
```

Comment:
get_agent_supported_features? the current name seems to imply it is CRP who supports those features

Reply:
The idea was exactly that, basically report status for all those features that CRP is aware of and the agent supports too lol. 
How about `get_agent_supported_feature_list_for_crp`?

Reply:
Done